src/main/java/gregtech/api/items/toolitem/IToolStats.java

```diff
@@ -175,10 +175,9 @@ default int getColor(ItemStack stack, int tintIndex) {
         return Collections.emptySet();
     }
 
-    default void onCraftingUse(ItemStack stack) {
-        if (ConfigHolder.client.toolCraftingSounds && ForgeHooks.getCraftingPlayer() != null && stack.getItem() instanceof ToolMetaItem<?>) {
-            EntityPlayer player = ForgeHooks.getCraftingPlayer();
-            player.getEntityWorld().playSound(player, player.getPosition(), ((ToolMetaItem<?>) stack.getItem()).getItem(stack).getSound(), SoundCategory.PLAYERS, 1, 1);
+    default void onCraftingUse(ItemStack stack, EntityPlayer player) {
+        if (ConfigHolder.client.toolCraftingSounds && player != null && stack.getItem() instanceof ToolMetaItem<?>) {
+            player.getEntityWorld().playSound(null, player.getPosition(), ((ToolMetaItem<?>) stack.getItem()).getItem(stack).getSound(), SoundCategory.PLAYERS, 1, 1);
```

Comment:
Should this be passing in `null` for the player when it was passing in the actual player before?

Reply:
Passing in `null` for the player causes this packet to be sent to every player on the server, while passing in a normal player as the argument sends it to everyone _except_ that player. Of course, that should not occur, so we forego putting a player into this argument, and just pass along its position.